### PR TITLE
Correct misspellings throughout the codebase

### PR DESCRIPTION
**Pull request by Sija**
SSIA / [misspel](https://github.com/client9/misspell) FTW.

### CODE REVIEW

File: spec/std/big/big_int_spec.cr [hunk 1]
```diff
@@ -56,7 +56,7 @@ describe "BigInt" do
     [1.1, 1.to_big_i, 3.to_big_i, 2.2].sort.should eq([1, 1.1, 2.2, 3])
   end
 
-  it "divides and calculs the modulo" do
+  it "divides and calculus the modulo" do
```
[the remark this answers is not in the document]
Comment (Sija) [reply spec/std/big/big_int_spec.cr]: or rather *calculates*.

**Comment by Sija** [comment]
CI failed with unrelated exception:

```
Unhandled exception in spawn: Error writing to socket: Broken pipe (Errno)
  from src/socket.cr:77:13 in 'unbuffered_write'
  from src/io/buffered.cr:123:14 in 'write'
  from src/io.cr:853:5 in 'write_byte'
  from src/io/buffered.cr:156:14 in 'write_byte'
  from src/char.cr:767:9 in 'to_s'
  from src/io.cr:184:5 in '<<'
  from src/http/client/response.cr:63:5 in 'to_io'
  from spec/std/http/client/client_spec.cr:13:7 in '->'
  from src/fiber.cr:255:3 in 'run'
  from src/fiber.cr:75:34 in '->'
  from ???
```

File: src/winerror.cr [hunk 2]
```diff
@@ -1817,7 +1817,7 @@ class WinError < Errno
   ERROR_SXS_XML_E_EXPECTINGCLOSEQUOTE                           = 14045_u32
   ERROR_SXS_XML_E_MULTIPLE_COLONS                               = 14046_u32
   ERROR_SXS_XML_E_INVALID_DECIMAL                               = 14047_u32
-  ERROR_SXS_XML_E_INVALID_HEXIDECIMAL                           = 14048_u32
```
Comment (oprypin) [inline src/winerror.cr]: This change is invalid.
https://docs.microsoft.com/en-us/windows/desktop/debug/system-error-codes--12000-15999-#error_sxs_xml_e_invalid_hexidecimal

Comment (Sija) [reply src/winerror.cr]: image

Comment (bew) [reply src/winerror.cr]: @oprypin so the Microsoft docs have a spelling error? Can't find anything explaining what hexidecimal could be (and if it's really a real word)

Comment (oprypin) [reply src/winerror.cr]: I think it's part of Windows API to have a constant with this misspelled name.

**Comment by Sija** [comment]
@straight-shoota Mind you there are several changes to the internal (`private` or `:nodoc:`) APIs, so not only docs are affected.

**Comment by straight-shoota** [comment]
That's why I also added the `kind:refactor` label.

**Pull request merged**